sale_coupon_advanced/models/sale_coupon_program.py

```diff
@@ -53,12 +51,11 @@ def _check_promo_code(self, order, coupon_code):
 
     @api.model
     def _filter_programs_from_common_rules(self, order, next_order=False):
-
         initial_programs = self.browse(self.ids)
         self._force_sale_order_lines(initial_programs, order)
         programs = super()._filter_programs_from_common_rules(order, next_order)
-        programs = programs._filter_first_order_programs(order)
-        programs = programs._filter_n_first_order_programs(order)
+        self._filter_order_programs(order, self._filter_first_order_programs)
```

Comment:
assigning programs here we trying to follow policy defined in the base method especially when we use a model decorator

Reply:
You mean, that I forgot to use ` programs` and used `self? Then yeah, will change that.